pkg/source/openstack/client.go

```diff
@@ -727,42 +703,44 @@ func (c *Client) findVM(name string) (*ExtendedServer, error) {
 	return &s, err
 }
 
-func (c *Client) ImageFirmwareSettings(instance *servers.Server) (bool, bool, bool, error) {
+func (c *Client) getFirmwareSettings(instance *servers.Server) (*source.Firmware, error) {
 	var imageID string
-	var uefiType, tpmEnabled, secureBoot bool
+	fw := source.NewFirmware(false, false, false)
```

Comment:
why not move `fw` decleration after the information has been looked up from the boot volume.

all scenarios where we return error before we can just `return nil, err`.

the error will be catched when this function is called and vm creation will not proceed anyways.